index/miner/miner.go

```diff
@@ -256,13 +307,43 @@ func (mi *MinerIndex) Close() error {
 	return nil
 }
 
-func getPower(ctx context.Context, c API, addr string) (*PowerInfo, error) {
+func getPower(ctx context.Context, c API, addr string) (Power, error) {
 	mp, err := c.StateMinerPower(ctx, addr, nil)
 	if err != nil {
-		return nil, err
+		return Power{}, err
 	}
-	return &PowerInfo{
-		Power:      mp.MinerPower.Uint64(),
-		TotalPower: mp.TotalPower.Uint64(),
+	p := mp.MinerPower.Uint64()
+	tp := mp.TotalPower
+	return Power{
+		Power:    p,
+		Relative: float64(p) / float64(tp.Uint64()),
 	}, nil
 }
+
+func getStaticInfo(ctx context.Context, c API, h *fchost.FilecoinHost, lr iplocation.LocationResolver, addr string) (Static, error) {
```

Comment:
Here's where the static info is gathered.
We connect to the miner byt its peerid. The real addr is resolved by our `FilecoinHost` instance... which is a peer in the FC network connected to the DHT. So all this resolution is magical from this side.
Regarding geo information, is resolved by `LocationResolver` abstraction... which the thing using the offline database for resolving multiaddrs.